### DISCUSSION TITLE
Allow whitelisting `.<key>` files and folders

### DISCUSSION THREAD

**Discussion by rclement** [vercel/vercel#4475] [General] [answered]
Hi Zeit/Vercel team,

As already confirmed with the support team, every `.<key>` files and folders are stripped from deployments for security reasons. Specifying `!.<key>` paths in `.nowignore` has no effect whatsoever.

I would like to be able to deploy specific `.<key>` files, as this is required for some specific use-cases:

- Keybase HTTPS proof: `/.well-known/keybase.txt`
- OpenPGP Web Key Directory: `/.well-known/openpgpkey/policy`, `/.well-known/openpgpkey/hu/<keyid>`

The proposed solution by the support team is to use `rewrites` in `now.json`, such as:
```json
{
  "rewrites": [
    { "source": "/.well-known/keybase.txt", "destination": "/keybase.txt" },
  ]
}
```

But this is not the way to go: this solution would require to change the structure of a deployment only to satisfy a specific decision from a cloud provider, and I strongly avoid "vendor lock-in" syndrome (typically, I often deploy to multiple static hosting providers for fail-over scenarios).

Given that `.<key>` files and folders are excluded by default, would it be possible to allow deployment of `.<key>` files using `.nowignore` whitelisting feature?

**Comment by styfle** [comment, marked as answer]
> every `.<key>` files and folders are stripped from deployments for security reasons

Not every dot file, only a few files should be ignored.
You can see the full list of exclusions here: https://vercel.com/docs/v2/build-step#ignored-files-and-folders

I can confirm this works correctly when using Next.js with a public directory as seen here: 
https://github.com/styfle/styfle.dev/blob/main/public/.well-known/keybase.txt

This looks like a bug with pure static deployments. I'll create a ticket, thanks!

**Reply by styfle** [reply]
@rclement This has been fixed 👍 

You can upload purely static files that begin with a dot such as `.well-known/keybase.txt` today, thanks!

**Reply by rclement** [reply]
Thanks for the support @styfle, truly appreciated! Just tested the fix, works like a charm!

**Comment by rclement** [comment]
Thanks @styfle, this is great news!
I'll be waiting for the fix impatiently!
